### PR TITLE
Add Chat service tests and CI

### PR DESCRIPTION
**Pull request by hiongkaihan**
Changes made:

1. Followed collab-service naming convention of socket routes
2. Added tests for chat-service
3. Added CI for chat-service

### CODE REVIEW

File: chat-service/package.json [hunk 1]
```diff
@@ -7,15 +7,18 @@
   "scripts": {
     "dev": "nodemon index.js",
     "start": "node index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "mocha --exit"
   },
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "chai": "^4.3.6",
```
Comment (ongweisheng) [inline chat-service/package.json:15]: chai and mocha should be npm i chai mocha --save-dev so that it will fall under devDependencies below

**Review by ongweisheng — CHANGES_REQUESTED** [review]

File: chat-service/package.json (hunk 1 above)
Comment (hiongkaihan) [reply chat-service/package.json]: Fixed

**Review by ongweisheng — APPROVED** [review]
LGTM!

File: frontend/src/components/InterviewPage.js [hunk 2]
```diff
@@ -25,8 +24,8 @@ const InterviewPage = () => {
     const username = location.state.username
 
     useEffect(() => {
-        socket.emit('join_room', { username, room: interviewId}); // Connect user to chat room
-        collabSocket.emit("JOIN", { roomId: interviewId })
+        chatSocket.emit('join', { username, room: interviewId }) // Connect user to chat socket room
+        collabSocket.emit("JOIN", { roomId: interviewId })       // Connect user to collab socket room
```
Comment (TheSpaceCuber) [inline frontend/src/components/InterviewPage.js:28]: Will it be confusing if the only differentiating factor is capital letters?

**Review by TheSpaceCuber — APPROVED** [review]
LGTM

File: frontend/src/components/InterviewPage.js (hunk 2 above)
Comment (hiongkaihan) [reply frontend/src/components/InterviewPage.js:28]: Not planning to differentiate them. "join" will mean to join that particular socket. I have asked Herrick to change to "join" as well and standardize his routes to non-capital letters.

Comment (hiongkaihan) [reply frontend/src/components/InterviewPage.js:28]: Maybe should standardize the naming of the parameters as well. chatSocket is using { room : interviewId } but collabSocket is using { roomId : interviewId }

Comment (TheSpaceCuber) [reply frontend/src/components/InterviewPage.js:28]: @Herrekt can acknowledge and resolve this convo

Comment (Herrekt) [reply frontend/src/components/InterviewPage.js:28]: Understood. Will do the required changes in the collab service test/ci or dockerizing PR.

**Review by Herrekt — APPROVED** [review]
Tested! LGTM

**Pull request merged**